src/test/java/pl/com/bottega/ecommerce/sales/domain/invoicing/BookKeeperTest.java

```diff
@@ -1,20 +1,133 @@
 package pl.com.bottega.ecommerce.sales.domain.invoicing;
-
-import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
 
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+import pl.com.bottega.ecommerce.canonicalmodel.publishedlanguage.ClientData;
+import pl.com.bottega.ecommerce.canonicalmodel.publishedlanguage.Id;
+import pl.com.bottega.ecommerce.sales.domain.productscatalog.Product;
+import pl.com.bottega.ecommerce.sales.domain.productscatalog.ProductBuilder;
+import pl.com.bottega.ecommerce.sales.domain.productscatalog.ProductType;
+import pl.com.bottega.ecommerce.sharedkernel.Money;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Random;
 
+@ExtendWith(MockitoExtension.class)
 class BookKeeperTest {
+    @Mock
+    private InvoiceFactory invoiceFactory;
+    private InvoiceRequest invoiceRequest;
+    @Mock
+    private TaxPolicy taxPolicy;
+    private ClientData clientData;
+    private BookKeeper bookKeeper;
 
     @BeforeEach
-    void setUp() throws Exception {
+    void init(){
+        bookKeeper = new BookKeeper(invoiceFactory);
+        clientData = new ClientData(Id.generate(),"Test Client");
+        invoiceRequest = new InvoiceRequest(clientData);
+    }
+
+    @Test
+    void invoiceWithOnePositionOnRequest() {
+        when(taxPolicy.calculateTax(any(ProductType.class),any(Money.class))).thenReturn(new Tax(Money.ZERO, "tax"));
+        when(invoiceFactory.create(clientData)).thenReturn(new Invoice(Id.generate(),clientData));
+
+        Product product = new ProductBuilder().withProductType(ProductType.FOOD).withPrice(new Money(2.50)).withName("Milk").build();
+        RequestItem requestItem = new RequestItemBuilder().withProductData(product).withQuantity(1).build();
+        invoiceRequest.add(requestItem);
+        int res = bookKeeper.issuance(invoiceRequest,taxPolicy).getItems().size();
+        assertEquals(1,res);
+    }
+
+    @Test
+    void invoiceWithCouplePositionsOnRequest(){
+        when(taxPolicy.calculateTax(any(ProductType.class),any(Money.class))).thenReturn(new Tax(Money.ZERO, "tax"));
+        when(invoiceFactory.create(clientData)).thenReturn(new Invoice(Id.generate(),clientData));
+
+        Product product1 = new ProductBuilder().withProductType(ProductType.FOOD).withPrice(new Money(2.50)).withName("Milk").build();
```

Comment:
kod siępowtarza - w builderze mogłyby być domyślne wartości, a tutaj tylko to co jest innego od domyślnych